torch/_inductor/codegen/cpp.py

```diff
@@ -1652,7 +1652,14 @@ def load(self, name: str, index: sympy.Expr):
                     self.disable_vec(f"{load_dtype} not loaded as float")
                 return var
 
-            if load_dtype not in self.load_supported_dtypes:
+            buffer = V.graph.get_buffer(name)
+            if (
+                load_dtype in [torch.int32, torch.int64]
+                and (buffer.data.layout.size.__len__() == 0)
```

Comment:
`not buffer.data.layout.size` or `len(buffer.data.layout.size) == 0`

Reply:
Thanks for the comments and changed.